### PR TITLE
Adding GA API version for Provisioning Services

### PR DESCRIPTION
**Pull request by kvish**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json [hunk 1]
```diff
@@ -0,0 +1,1897 @@
...
+         "type": "string",
+         "maxLength": 256
+        },
+        {
+         "name": "certificateDescription",
+         "in": "body",
+         "description": "The certificate body.",
+         "required": true,
+         "schema": {
+          "$ref": "#/definitions/CertificateBodyDescription"
```
Comment (jianghaolu) [inline specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable.

File: specification/provisioningservices/resource-manager/readme.md [hunk 2]
```diff
@@ -26,16 +26,16 @@ These are the global settings for the API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2017-08
```
Comment (jianghaolu) [inline specification/provisioningservices/resource-manager/readme.md]: Is removing the previous tag intentional?

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSCreate.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          "etag": "AAAAAAAADGk=",
+          "name": "myFirstProvisioningService",
+          "location": "eastus",
+          "properties": {
+            "state": "Active",
+            "allocationPolicy": "Hashed",
+            "serviceOperationsHostName": "myFirstProvisioningService.azure-devices-provisioning.net",
+            "deviceProvisioningHostName": "global.azure-devices-provisioning.net",
+            "idScope": "0ne00000012",
+            "authorizationPolicies": []
```
Comment (rajeevmv) [inline specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSCreate.json]: Should have atleast one  'provisioningServiceOwner' rule.

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSDelete.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+    "parameters": {
+        "provisioningServiceName": "myFirstProvisioningService",
+        "resourceGroupName": "myResourceGroup",
+        "api-version": "2017-11-15",
+        "subscriptionId": "91d12660-3dec-467a-be2a-213b5544ddc0"
+    },
+    "responses": {
+      "200": {},
```
Comment (rajeevmv) [inline specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSDelete.json]: >  "200": {}, [](start = 4, length = 12)

when do we return 200? If there is no content, it should be 204.

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSCreate.json (hunk 3 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSCreate.json]: this is autocreated if not passed

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json [hunk 5]
```diff
@@ -0,0 +1,1897 @@
...
+          "$ref": "#/definitions/ErrorDetails"
+         }
+        }
+       },
+       "x-ms-pageable": {
+        "nextLinkName": "nextLink"
+       }
+      }
+     },
+     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/provisioningServices/{provisioningServiceName}/certificates/{certificateName}": {
```
[the remark this answers is not in the document]
Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: yes this is defined on line 595: "operationId": "DpsCertificates_List"

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json (hunk 1 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: the request and response are the same but the CertificateResponse includes some system generated properties that are added to CertificateBodyDescription which is why it has this format

File: specification/provisioningservices/resource-manager/readme.md (hunk 2 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/readme.md]: unintentional will fix this

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json (hunk 1 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: the PUT is a subset of the GET

**Comment by kvish** [comment]
Looking for review follow up

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSDelete.json (hunk 4 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/examples/DPSDelete.json]: This is returned as a response to the status polling request for the delete operation. The body contains the resource representation that indicates a transitional provisioning state.

File: specification/provisioningservices/resource-manager/readme.md [hunk 6]
```diff
@@ -29,15 +27,29 @@ openapi-type: arm
...
 ```
 
+``` yaml
+openapi-type: arm
+tag: package-2017-11
+```
+
 ### Tag: package-2017-08
 
 These settings apply only when `--tag=package-2017-08` is specified on the command line.
```
Comment (jhendrixMSFT) [inline specification/provisioningservices/resource-manager/readme.md]: Description text is out of sync with yaml.

File: specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json [hunk 7]
```diff
@@ -0,0 +1,1917 @@
...
+        }
+       }
+      }
+     },
+     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/provisioningServices/{provisioningServiceName}/certificates": {
+      "get": {
+       "tags": [
+        "GET"
+       ],
+       "operationId": "DpsCertificates_List",
```
Comment (jhendrixMSFT) [inline specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: All the other operation groups are named DpsCertificate (no ending 's'), presumably this is a type-o?

Comment (kvish) [reply specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json]: this is off a different controller called Certificates not certificate which us where it gets its name

File: specification/provisioningservices/resource-manager/readme.md [hunk 8]
```diff
@@ -29,15 +27,29 @@ openapi-type: arm
...
 
-``` yaml $(tag) == 'package-2017-08'
+``` yaml $(tag) == 'package-2017-11'
 input-file:
 - Microsoft.Devices/2017-08-21-preview/iotdps.json
```
Comment (jhendrixMSFT) [inline specification/provisioningservices/resource-manager/readme.md]: The tag says 2017-11 but the directory is 2017-08-21-preview which I believe is causing the linter failure.

File: specification/provisioningservices/resource-manager/readme.md (hunk 6 above)
Comment (kvish) [reply specification/provisioningservices/resource-manager/readme.md]: resolved

**Comment by kvish** [comment]
Note on the following linter errors: 

{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'Code', for definition: 'ErrorDetails' must follow camelCase style. Example: 'code'.",
  "id": "R3016",
  "validationCategory": "RPCViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json:1687:10 ($.definitions.ErrorDetails.properties.Code)"
  ]
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'HttpStatusCode', for definition: 'ErrorDetails' must follow camelCase style. Example: 'httpStatusCode'.",
  "id": "R3016",
  "validationCategory": "RPCViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json:1692:10 ($.definitions.ErrorDetails.properties.HttpStatusCode)"
  ]
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'Message', for definition: 'ErrorDetails' must follow camelCase style. Example: 'message'.",
  "id": "R3016",
  "validationCategory": "RPCViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json:1697:10 ($.definitions.ErrorDetails.properties.Message)"
  ]
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'Details', for definition: 'ErrorDetails' must follow camelCase style. Example: 'details'.",
  "id": "R3016",
  "validationCategory": "RPCViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/provisioningservices/resource-manager/Microsoft.Devices/2017-11-15/iotdps.json:1702:10 ($.definitions.ErrorDetails.properties.Details)"
  ]
our Resource Provisioning Services is currently tied to out other resource Iot Hubs which uses this casing. We have a backlog item to update this casing across all our resources

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**